[Ryan Chua Zong Xun] iP

ryanczx opened the pull request:
# FredPro
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

FredPro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/ryanczx/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ryanczx committed:
changed file name and class name to Fred, did Greet, Echo, and Exit

ryanczx committed:
changed the main conditional to a switch statement

ryanczx committed:
added Add and List

ryanczx committed:
created new Task class and adapted the code in Fred.java for it

ryanczx committed:
updated Task class with status icon

ryanczx committed:
changed switch statement to if-else statement

ryanczx committed:
made list method of Fred more straight forward

ryanczx committed:
added Mark as Done

ryanczx committed:
added ToDo, Event, and Deadline, and improved Task

ryanczx committed:
added the missing chatbot dialogue for Level-3 tasks

ryanczx committed:
Added logic for ToDo, Event, and Deadline. Improved mark and unmark to support more than one digit. Also fixed minor error in event.

ryanczx committed:
updated bash script for testing, updated input.txt and EXPECTED.TXT, removed one line from Fred to make testing easier

ryanczx committed:
added FredException class and the error handling in Fred

ryanczx committed:
added the basic logic for Delete

ryanczx committed:
added error handling for Delete and when input int is lesser than 1

ryanczx committed:
Add Save and Load

ryanczx committed:
Add Dates

ryanczx committed:
Merge branch 'branch-Level-7'

ryanczx committed:
Merge branch 'branch-Level-8'

ryanczx committed:
Fix Dates problem due to merge commit 'Level-8'

ryanczx committed:
Refactor the code

ryanczx committed:
Organise classes into packages

ryanczx committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

ryanczx committed:
Edit build.gradle so that it can build and run chatbot

ryanczx committed:
Add JUnit tests

ryanczx committed:
Create JAR file

ryanczx committed:
Add JavaDoc comments

ryanczx committed:
Change code to follow coding standard

ryanczx committed:
Add Find

ryanczx committed:
Merge branch 'branch-A-JavaDoc'

ryanczx committed:
Merge branch 'branch-A-CodingStandard'

ryanczx committed:
Merge branch 'branch-Level-9'

ryanczx committed:
Add JavaDocs for new class and method related to Find

Rachel-AG commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,20 @@
+import tasklist.TaskList;
+
+import ui.Ui;
+
```
I think it would be better if you add a header here

Rachel-AG commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,20 @@
+    protected boolean isExit;
+
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws FredException;
+
```
Perhaps you can add a header here so you can simply inherit the javadocs for DeleteCommand, AddCommand, etc

Rachel-AG commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,122 @@
+package parser;
+
+import commands.*;
```
Perhaps you can consider explicitly listing the imported classes for the commands package?

Rachel-AG reviewed (commented):
Nice! Generally, I think you have followed the coding standards closely :)

CedricChia123 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,122 @@
+            throw new FredException("I'm sorry, but I don't know what that means :(");
+        }
+    }
+}
```
The method for parse seems to be very long. Perhaps you can split the checks for commands up?

CedricChia123 commented on src/main/java/commands/AddCommand.java:
```diff
@@ -0,0 +1,58 @@
+            tasks.add(new Deadline(description, date));
+            break;
+        }
+
```
Maybe could add some javadocs for this method, and any other methods too

CedricChia123 commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,122 @@
+
+            int index;
+            try {
+                index = Integer.parseInt(command.substring(7));
```
Perhaps the use of magic numbers could be changed

CedricChia123 reviewed (commented):
In general, the code is clean and easy to follow, with great OOP principles used. Some methods are missing javadocs and could be shortened in length. Other than that, great job!

ryanczx committed:
Parser.java: List out all command imports

ryanczx committed:
Command.java: Add JavaDocs for Command class

ryanczx committed:
Setup CheckStyle and fix CheckStyle violations

ryanczx committed:
Add a line to build.gradle file (useless commit, I forgot to make branch)

ryanczx committed:
Delete a line from build.gradle file (useless commit, I forgot to make branch)

ryanczx committed:
Merge branch 'branch-A-CheckStyle'

ryanczx committed:
Get basic JavaFX working with Fred

ryanczx committed:
Get Fred functionality working with JavaFX

ryanczx committed:
Merge branch 'branch-Level-10'

ryanczx committed:
Update build.gradle so that Gradle can generate JAR files

ryanczx committed:
Add Assertions for Fred.java and Storage.java

For Fred.java, add an Assertion to assert that Fred has generated a
non-empty message after receivng input from the user.

For Storage.java, add an Assertion in save method to assert that the
data file exists so that Fred can save tasks into the data file. Also
add an Assertion in load method to assert that the program only reaches
the else block if a line in the data file does not start with either
"T", "E" or "D".

ryanczx committed:
Ui.java: remove unused fields and methods, and add javadocs

ryanczx committed:
Parser.java: extract methods, and fix Fred welcome

Parser.java: extract methods from parse method conditional.

MainWindow.java: Fix Fred welcome message.

ryanczx committed:
Storage.java: extract methods

Storage.java: extract the methods from load method in Storage

ryanczx committed:
Parser.java fix javadocs

ryanczx committed:
Parser.java: change name of extracted methods

ryanczx committed:
Improve parser by removing magic numbers

ryanczx committed:
Fix minor error in parseFindCommand

ryanczx committed:
Fix ExitCommand so that it exits the program

ryanczx committed:
Improve load method in Storage.java by making code more readable

ryanczx committed:
Parser.java change access modifier of extracted methods to private

ryanczx committed:
Move the parsing of dateString into date into extracted methods for load method inside Storage.java

ryanczx committed:
Merge pull request #1 from ryanczx/A-Assertions

Add Assertions for Fred.java and Storage.java

ryanczx committed:
Merge branch 'master' into A-CodeQuality

ryanczx committed:
Merge pull request #2 from ryanczx/A-CodeQuality

A-CodeQuality into master

ryanczx committed:
Storage.java: fix the assert in load method

ryanczx committed:
Add gradle.yml file

ryanczx committed:
gradle.yml: delete I/O redirection tests

ryanczx committed:
Fix checkstyle errors

ryanczx committed:
Fix checktyle errors in FredTest.java

ryanczx committed:
Add snooze command to reschedule tasks

ryanczx committed:
Merge branch 'branch-B-Snooze'

ryanczx committed:
Add green background to Fred dialog and blue background to User dialog

ryanczx committed:
Add padding to dialog boxes text

ryanczx committed:
Set dialog box height to be not fixed for longer messages

ryanczx committed:
Increase side padding for dialog box text

ryanczx committed:
Fix import ordering

ryanczx committed:
Merge branch 'branch-A-BetterGui'

ryanczx committed:
Remove unused delimiter from scanner in load method

ryanczx committed:
Add title to GUI window

ryanczx committed:
Add Ui.png to docs

ryanczx committed:
Change find command to show actual index of tasks

Change the find command to show the index of tasks where the index of
each task would be the index that the task had if the list command were
to be run.

ryanczx committed:
README.md update user guide

ryanczx committed:
README.md: improve user guide

Add linebreaks, screenshot of GUI and acknowledgements section.

ryanczx committed:
README.md: more improvements

Add list of features with links within the user guide and reduce size of
screenshot.

ryanczx committed:
README.md: Fix broken links in feature list

ryanczx committed:
README.md Add link back to Features for each feature header

ryanczx committed:
README.md: Remove broken links back to Feature

ryanczx committed:
Fix javaFX version number in FXML files

ryanczx committed:
Fix load and save

Change the logic to show a message in the GUI on startup whether data file in hard
disk has been loaded successfully.

Change the filepaths to use java.nio.file.Path instead of using String.

Fix load() method to create data directory if it does not exist.

ryanczx committed:
Fred.java: Update javadocs for getResponse() method

ryanczx committed:
Change generated jar file name

ryanczx committed:
Storage.java: Simplify save method
